### PR TITLE
feature: include ying-yang dataset

### PR DESCRIPTION
**Pull request by cpehle**
Adds the Ying-Yang dataset to the norse code base. Intention is to quickly be able to demonstrate time-to-first spike learning with the Norse library.

### CODE REVIEW

File: norse/dataset/test/test_ying_yang.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from norse.dataset.yingyang import YinYangDataset
+
+def test_generation():
+    dataset_train = YinYangDataset(size=20, seed=42)
+    dataset_validation = YinYangDataset(size=30, seed=41)
+    dataset_test = YinYangDataset(size=40, seed=40)
```
Comment (Jegp) [inline norse/dataset/test/test_ying_yang.py:6]: Perhaps add an assert on the shape, just for posterity?

File: norse/dataset/yingyang/__init__.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+Initializing the dataset:
+
+.. code: python
+    from norse.dataset.yingyang import YinYangDataset
+
+    dataset_train = YinYangDataset(size=5000, seed=42)
+    dataset_validation = YinYangDataset(size=1000, seed=41)
+    dataset_test = YinYangDataset(size=1000, seed=40)
+
+**Note** It is very important to give different seeds for trainings-, validation- and test set, as the data is generated randomly using rejection sampling. Therefore giving the same seed value will result in having the same samples in the different datasets!
```
Comment (Jegp) [inline norse/dataset/yingyang/__init__.py:41]: An approach to the random seed problem I've seen several times would be to leave the seed unset (default param `None`) to default to random values. In that way ignorant/busy users won't accidentally reuse the seed.
I think I would prefer that approach to trusting users to read the docs (yeah, right) and correctly set the parameter.

**Review by Jegp — CHANGES_REQUESTED** [review]
Neat addition, good docs.

I would like to address/discuss the random seed issue, otherwise we're good to go!

File: norse/dataset/yingyang/__init__.py (hunk 2 above)
Comment (cpehle) [reply norse/dataset/yingyang/__init__.py:41]: Yeah that makes sense :).

**Comment by Jegp** [comment]
I'll suggest adding this to the [Tonic library](https://tonic.readthedocs.io/)

**Pull request closed without merging**